index.d.ts

```diff
@@ -1,123 +1,658 @@
 }
 
-declare class Path{
-    api: SwaggerApi
-    definition: Object
-    definitionFullyResolved: Object
-    operationObjects: Operation[]
-    parameterObjects: Parameter[]
-    path: string
-    pathToDefinition: string[]
-    ptr: string
-    regexp: RegExp
-
-    getOperation(method : string): Operation[]
-    getOperations(): Operation[]
-    getOperationsByTag(tag): Operation[]
-    getParameters(): Parameter[]
+/**
+ * The Path object.
+ *
+ * Extra Properties: Other than the documented properties, this object also exposes all properties of the
+ * definition object.
+ */
+declare class Path {
+  /** The SwaggerApi object */
+  api: SwaggerApi
+
+  /**
+   * The path definition (The raw path definition after remote references were resolved)
+   */
+  definition: Object
+
+  /**
+   * The path definition with all of its resolvable references resolved
+   */
+  definitionFullyResolved: Object
+
+  /**
+   * The `Operation` objects
+   */
+  operationObjects: Operation[]
+
+  /**
+   * The path-level `Parameter` objects
+   */
+  parameterObjects: Parameter[]
+
+  /**
+   * The path string
+   */
+  path: string
+
+  /**
+   * The path segments to the path definition
+   */
+  pathToDefinition: string[]
+
+  /**
+   * The JSON Pointer to the path
+   */
+  ptr: string
+
+  /**
+   * The `RegExp` used to match request paths against this path
+   */
+  regexp: RegExp
+
+  /**
+   * The Path object.
+   *
+   * Note: Do not use directly.
+   * @param api The SwaggerApi object
+   * @param path The path string
+   * @param definition The path definition (The raw path definition after remote references were resolved)
+   * @param definitionFullyResolved The path definition with all of its resolvable references resolved
+   * @param pathToDefinition 	The path segments to the path definition
+   */
+  new(api: SwaggerApi, path: string, definition: Object, definitionFullyResolved: Object, pathToDefinition: string[]): Path
+
+  /**
+   * Return the operation for this path and method.
+   * @param method
+   * @return The `Operation` objects for this path and method or undefined if there is no operation for the provided method
+   */
+  getOperation(method: string): Operation[]
+
+  /**
+   * Return the operations for this path.
+   * @return {Operation[]} The `Operation` objects for this path
+   */
+  getOperations(): Operation[]
+
+  /**
+   * Return the operations for this path and tag.
+   * @param tag the tag
+   * @return {Operation[]} The `Operation` objects for this path and tag
+   */
+  getOperationsByTag(tag: string): Operation[]
+
+  /**
+   * Return the parameters for this path.
+   * @return The `Parameter` objects for this path
+   */
+  getParameters(): Parameter[]
 }
 
-declare class ParameterValue{
-    error: Error
-    parameterObject: Parameter
-    raw: Object
-    valid: boolean
-    value: Object
-}
\ No newline at end of file
+
+/**
+ * Object representing a parameter value.
+ */
+export declare class ParameterValue {
```

Comment:
export `ParameterValue` so it can be used to type `swagger-node-runner` middleware.